src/main/java/com/google/sps/data/PrioritizeCoursesCriteria.java

```diff
@@ -0,0 +1,41 @@
+package com.google.sps.data;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Arrays;
+import java.util.ArrayList;
+
+public class PrioritizeCoursesCriteria implements Preference {
+  List<Course> courseList;
+  
+  public PrioritizeCoursesCriteria(List<Course> courseList) {
+    this.courseList = courseList;
+  }
+  
+  /**
+   * Implements the preference score of the schedule, based on the criteria
+   * that prioritizes the order of courses that the user would like to take
+   */
+  public float preferenceScore(Schedule schedule) {
+    List<String> courseListNames = getCourseListNames(courseList);
+    int coursePrioritySum = 0;
+    List<Course> courses = (List) schedule.getCourses();
+    
+    for (Course course : courses) {
+      String courseName = course.getName();
+      int priorityIndex = courseListNames.size() - courseListNames.indexOf(courseName);
```

Comment:
`courseListNames.indexOf(...)` is an O(N) operation you're going to repeat for every course you want to score.

It would be both more efficient and simple if instead, the code...
1. Created a `HashMap<String, Integer> courseScores` member in this class
2. Instantiated and filled in all of the values in this map in your constructor
3. Did a O(1) lookup in this map to find the course scores when adding them up

Reply:
That's a good idea. Only thing I changed was defining HashMap with <Course, Integer>